prov/efa/src/efa_av.c

```diff
@@ -317,7 +317,17 @@ int efa_conn_rdm_init(struct efa_av *av, struct efa_conn *conn)
 			return err;
 		}
 
-		ret = fi_av_insert(av->shm_rdm_av, smr_name, 1, &peer->shm_fiaddr, 0, NULL);
+		/*
+		 * The shm provider supports FI_AV_USER_ID flag. This flag
+		 * associates a user-assigned identifier with each av entry that is
+		 * returned with any completion entry in place of the AV's address.
+		 * EFA provider could use efa's fi_addr as the user id (passed as
+		 * as an input to the *fi_addr argument in fi_av_insert call)
+		 * With this change, there is no need to translate shm address back 
+		 * to efa address when reading shm cq entry.
+		 */
+		peer->shm_fiaddr = conn->fi_addr;
```

Comment:
should we remove `shhm_fiaddr` from `efa_rdm_peer`?

Reply:
No, we still need it. The `peer->shm_fiaddr` returned by `fi_av_insert` is still the addr assigned by shm provider, which could be different from efa_fi_addr. efa code should still use `peer->shm_fiaddr` for shm sendmsg. 

The user id will only show in the cq_entry->addr. I will add comments to explain this.

Reply:
Added comment in latest revision.